### PR TITLE
Added text to be displayed when invalid data is entered into search input field and  no search results are found

### PR DESCRIPTION
**Pull request by san-coding**
Added text to be displayed when invalid data is entered into the search input field and no search results found
 in `WooCommerce-> Marketplace` , fixes issue #30588

### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Closes # .

### How to test the changes in this Pull Request:

1. Create any test site using JN site.
2. Install and activate all the required plugins.
3. Install and activate WooCommerce 5.7.0 Beta-1.
4. Complete the setup wizard.
5. Go to WooCommerce->Marketplace.
6. Search invalid data (non-existing) in search field.
7. Screen displays `Sorry, could not find anything. Try searching again using a different term.`

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Tweak - Show a search again message when marketplace results are empty.

### FOR PR REVIEWER ONLY:

* [x] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Comment by san-coding** [comment]
Hey @vedanshujain @andfinally @kloon  , I have added this PR #30641 , please review it , thank you

**Comment by san-coding** [comment]
Ok , now I see that the changes I have made are wrong, but how do I know that no search results are being returned

**Comment by san-coding** [comment]
> now I see that the changes I have made are wrong, but how do I know that no search results are being returned , is any variable storing the number of results returned, please help me out @vedanshujain @andfinally @kloon , thanks

**Comment by vedanshujain** [comment]
@san-coding you should be able to use `$addon` to see if there are search results. Can you check if the presence of both `$search` and more than one in `$addon` gives you expected behavior?

**Comment by san-coding** [comment]
> @san-coding you should be able to use `$addon` to see if there are search results. Can you check if the presence of both `$search` and more than one in `$addon` gives you expected behavior?

Thanks @vedanshujain , I have made the changes , instead of checking `if(empty($search))` , I am now checking `if(sizeof($addons)==0)`

I have also added <?php if ( ! empty( $search )`&& sizeof($addons)>0` ) : ?> , so that the message that the message **search results for ""** is displayed only when `&& sizeof($addons)>0`

please verify and review if my changes are correct, thanks @vedanshujain

**Comment by san-coding** [comment]
Hey @vedanshujain , could you please review this

**Comment by san-coding** [comment]
Have added required space and replaced `sizeof()` with `count()`

**Comment by san-coding** [comment]
@vedanshujain , looks like all checks have passed, you may review and merge

**Comment by san-coding** [comment]
@vedanshujain @rodelgc @andfinally someone please review this

File: includes/admin/views/html-admin-page-addons.php
Comment (kloon) [inline includes/admin/views/html-admin-page-addons.php]: You can remove this since there is no placeholder in the string below.

**Review by kloon — CHANGES_REQUESTED** [review]

File: includes/admin/views/html-admin-page-addons.php
Comment (san-coding) [reply includes/admin/views/html-admin-page-addons.php:67]: should I make the below change ?
```suggestion
					<?php // translators: search keyword. ?>
```

Comment (san-coding) [reply includes/admin/views/html-admin-page-addons.php]: What exactly should I remove

Comment (eopws) [reply includes/admin/views/html-admin-page-addons.php]: He is talking about this `<?php // translators: search keyword. ?>`

**Comment by san-coding** [comment]
Thank you , I have made the changes, please review this PR

**Comment by san-coding** [comment]
can you please add a label `hacktoberfest-accepted` to this PR, thank you @eopws @kloon

File: includes/admin/views/html-admin-page-addons.php
Comment (san-coding) [reply includes/admin/views/html-admin-page-addons.php]: have made the changes

**Comment by san-coding** [comment]
@kloon @eopws @vedanshujain please review this

**Review by kloon — APPROVED** [review]
Changes look good to me now, thanks!

**Comment by san-coding** [comment]
@kloon thanks

**Comment by peterfabian** [comment]
Requesting a review from the core team before merging just to double check, should be a quick one.

**Review by vedanshujain — APPROVED** [review]
LGTM!

**Comment by san-coding** [comment]
Thanks you 👍🏻

**Comment by peterfabian** [comment]
> A pull request is considered approved once it has an overall approving review from maintainers, or has been merged by maintainers, or has been given the 'hacktoberfest-accepted' label.

My reading of [the rules](https://hacktoberfest.digitalocean.com/) is that the label is not necessary, but please let us know if you think we need to add it. Thx!

**Comment by drjamesj** [comment]
Hi @san-coding @vedanshujain - there is a slight issue for this PR when there is no search term the error message still displays (ie. on the landing page when you first access Marketplace).

To reproduce:
- login to wp-admin
- navigate to WooCommerce -> Marketplace
- you will see the error message `Sorry, could not find anything. Try searching again using a different term.` on this page when it should not show here

image

I am happy to open a PR to correct this if you like. Just let me know.

**Comment by san-coding** [comment]
Ohh i will look into this and get back to you soon, can you revert the merge, adding a condition to check if search term exists would be enough i guess

**Comment by san-coding** [comment]
@drjamesj , checkout this file in the trunk branch, maybe it has been solved
https://github.com/woocommerce/woocommerce/blob/7486c0540a111aa518dfbfdd93484b5791b2ecbd/includes/admin/views/html-admin-page-addons.php#L67

**Pull request merged**
